### PR TITLE
split icds reports

### PR DESCRIPTION
**Pull request by emord**
Adds the ability to have one pillow per UCR table_id and able to exclude ucrs based on table id

@snopoke @sravfeyn @sheelio

### CODE REVIEW

**Comment by emord** [comment]
updated to fix tests and include a list of ucrs and selectively listen to kafka topics

File: corehq/apps/userreports/pillow.py [hunk 1]
```diff
@@ -29,19 +29,39 @@
...
+    def __init__(self, data_source_provider, auto_repopulate_tables=False, *args, **kwargs):
         self.bootstrapped = False
         self.last_bootstrapped = datetime.utcnow()
         self.data_source_provider = data_source_provider
         self.auto_repopulate_tables = auto_repopulate_tables
-        self.ucr_start = ucr_division[0]
-        self.ucr_end = ucr_division[-1]
+        self.ucr_division = kwargs.pop('ucr_division', None)
+        self.include_ucrs = kwargs.pop('include_ucrs', None)
+        self.exclude_ucrs = kwargs.pop('exclude_ucrs', None)
```
Comment (snopoke) [inline corehq/apps/userreports/pillow.py]: Could be good to raise an error if both include and division are specified.

**Review by snopoke — APPROVED** [review]
Nice this is great.

File: corehq/apps/userreports/pillow.py (hunk 1 above)
Comment (emord) [inline corehq/apps/userreports/pillow.py]: good idea https://github.com/dimagi/commcare-hq/pull/15479/commits/9890aafdfb7d07bc13502c87567d76aa13cfdc3f

File: corehq/apps/userreports/pillow.py [hunk 2]
```diff
@@ -27,21 +27,47 @@
...
 
+class PillowConfigError(Exception):
+    pass
+
+
 class ConfigurableReportTableManagerMixin(object):
 
-    def __init__(self, data_source_provider, auto_repopulate_tables=False,
-                 ucr_division='0f', *args, **kwargs):
+    def __init__(self, data_source_provider, auto_repopulate_tables=False, *args, **kwargs):
```
Comment (sravfeyn) [inline corehq/apps/userreports/pillow.py]: Would be nice to have a docstring for `ucr_division` kwarg

**Comment by sravfeyn** [comment]
Nice! Not merging because of conflicts

**Pull request merged**